lib/lotus/entity.rb

```diff
@@ -235,6 +267,30 @@ def update(attributes={})
       end
     end
 
+    # Getter for hash of changed attributes.
+    # Return empty hash, if there is no changes
+    # Getter for hash of changed attributes. Value in it is the previous one.
+    #
+    # @return [::Hash] the changed attributes
+    #
+    # @since x.x.x
+    #
+    # @example
+    #   require 'lotus/model'
+    #   class Article
+    #     include Lotus::Entity
+    #     attributes :title
+    #   end
+    #
+    #   article = Article.new(title: 'The crime and punishment')
+    #   article.changed_attributes # => {}
+    #   article.title = 'Master and Margarita'
+    #   article.changed_attributes # => {:title => 'The crime and punishment'}
+    #
+    def changed_attributes
+      @changed_attributes ||= Hash.new
```

Comment:
This isn't thread safe, this ivar should be set when the entity is initialized. Then the getter should be:

```ruby
def changed_attributes
  @changed_attributes.dup
end
```

In this way we prevent accidental modifications of the internal state of the object. Here an example:

```ruby
book.changed_attributes # => {:title => "The crime and punishment"}
book.changed_attributes.delete(:title)
book.changed_attributes # => {}
```